src/frontend/src/session/cursor_manager.rs

```diff
@@ -144,36 +144,44 @@ pub struct SubscriptionCursor {
 impl SubscriptionCursor {
     pub async fn new(
         cursor_name: String,
-        start_timestamp: i64,
-        from_snapshot: bool,
+        start_timestamp: Option<i64>,
         subscription: Arc<SubscriptionCatalog>,
         handle_args: &HandlerArgs,
     ) -> Result<Self> {
-        let state = if from_snapshot {
+        let state = if let Some(start_timestamp) = start_timestamp {
+            State::InitLogStoreQuery {
+                seek_timestamp: start_timestamp,
+                expected_timestamp: None,
+            }
+        } else {
             // The query stream needs to initiated on cursor creation to make sure
             // future fetch on the cursor starts from the snapshot when the cursor is declared.
             //
             // TODO: is this the right behavior? Should we delay the query stream initiation till the first fetch?
-            let (row_stream, pg_descs) = Self::initiate_query(
-                from_snapshot,
-                start_timestamp,
-                &subscription,
-                handle_args.clone(),
-            )
-            .await?;
+            let (row_stream, pg_descs) =
+                Self::initiate_query(None, &subscription, handle_args.clone()).await?;
+            let pinned_epoch = handle_args
+                .session
+                .get_pinned_snapshot()
+                .ok_or_else(|| {
+                    ErrorCode::InternalError("Fetch Cursor can't find snapshot epoch".to_string())
+                })?
+                .epoch_with_frontend_pinned()
+                .ok_or_else(|| {
+                    ErrorCode::InternalError(
+                        "Fetch Cursor can't support setting an epoch".to_string(),
+                    )
+                })?
+                .0;
+            let start_timestamp = convert_epoch_to_logstore_i64(pinned_epoch);
```

Comment:
Please add some comments on why we need to get pinned epoch after `initiate_query`.